core/crypto/type_hasher.hpp

```diff
@@ -24,12 +24,12 @@ namespace kagome::crypto {
   struct Hashed {
     static_assert(N == 8 || N == 16 || N == 32 || N == 64,
                   "Unexpected hash size");
-    using Type = std::decay_t<T>;
+    static_assert(std::is_same_v<T, std::decay_t<T>>);
+    using Type = T;
     using HashType = common::Blob<N>;
 
    public:
-    template <typename... Args>
-    Hashed(Args &&...args) : type_{std::forward<Args>(args)...} {}
+    explicit Hashed(Type type) : type_{std::move(type)} {}
```

Comment:
зачем?

Reply:
`static_assert` checks that type is not reference (to avoid dangling references, may use `shared_ptr` instead).

`auto &&` constructor was ambigous and conflicted with copy/move constructor.

Reply:
тип задумывался чтобы быть бесплатным, как в случае вычисления хэша, там и в случае передачи объекта, теперь он требует обязательного копирования. Я не хочу использовать shared_ptr когда это не нужно, например если объект существует на стеке, а использование reference_wrapper не убережет от висячих ссылок. Revert plz.

Reply:
fixed